### PR TITLE
tests: data server component unit tests

### LINKED ISSUE #521: [TEST] Jax Data Server
### What do you want to test?
Jax data server components

### Outline of test structure
* Unit tests
* Test components and hooks

### Definition of done
Passing checks, cover all hooks, edge cases considered

### Mandatory checklist before making a PR
* [ ] The success criteria laid down in “Definition of done” are met.
* [ ] Test code is documented - docstrings for methods and classes, static types for arguments.
* [ ] Documentation is updated - README, CONTRIBUTING, or other documentation.

### PR DESCRIPTION
**Pull request by RuanJohn**
## What?
Unit tests for `DataServer` components.
## Why?
Part of expanding test code coverage for callback redesign Jax-based Mava.  
## Extra
Existing dataserver tests were moved to `tests/jax/integration` since they depended on multiple components. We will utilize there tests as part of the integration tests. 
These unit tests are simplified versions of the integration tets that don't require a components to be passed to a system and a system to be built. 
closes #521

### CODE REVIEW

File: tests/jax/components/building/data_server_test.py [hunk 1]
```diff
@@ -13,90 +13,162 @@
...
+    mock_builder.store.adder_signature_fn = lambda env_specs, extras_specs: reverb_adders.ParallelNStepTransitionAdder.signature(  # noqa: E501
+        env_specs, extras_specs
+    )
+
+    data_server = OnPolicyDataServer()
+    data_server.on_building_data_server(mock_builder)
+
+    table = mock_builder.store.data_tables[0]
+
+    assert table.info.max_size == 1000
```
Comment (OmaymaMahjoub) [inline tests/jax/components/building/data_server_test.py:154]: I think we need to verify the name and signature here of the table

File: tests/jax/components/building/data_server_test.py [hunk 2]
```diff
@@ -13,90 +13,162 @@
...
+    )
+
+    mock_builder.store.sequence_length = 20
+
+    data_server = OnPolicyDataServer()
+    data_server.on_building_data_server(mock_builder)
+
+    table = mock_builder.store.data_tables[0]
+
+    assert table.info.max_size == 1000
```
Comment (OmaymaMahjoub) [inline tests/jax/components/building/data_server_test.py:174]: same here

**Review by OmaymaMahjoub — CHANGES_REQUESTED** [review]
I had a look at the unit test and it is good just two missing things to verify also the integration can't be in another branch? or at least we need to change the PR name because this PR is for a unit test from what I think?

File: tests/jax/components/building/data_server_test.py (hunk 2 above)
Comment (RuanJohn) [reply tests/jax/components/building/data_server_test.py:176]: Sorted.

**Review by OmaymaMahjoub — APPROVED** [review]
Looks good! Thanks @RuanJohn

**Review by AsadJeewa — APPROVED** [review]
Thanks @RuanJohn! Clean code and you covered all the different params/ configs
